### ISSUE TITLE
Incorrect block ID's in overlays (and a few other overlay issues)

### ISSUE THREAD

**Issue by Toatrika** [xalixilax/xali-s-Enhanced-Vanilla-16x#21]
I found some incorrect block IDs in the overlays. Some of the things listed might of course be me missing some sort of functionality, but to my understanding only actual block ID's used by Minecraft work here. I have of course tried to see if incorrect ID's work in-game anyway and they do not (at least for me).

1. `basalt`
   - `grass_path` should be `dirt_path`
   - `silver_glazed_terracotta` does not exist
   - `sandstone_stairs` do exist, but cause clipping; additionally, no other stairs are mentioned in `basalt`

2. `bricks`
   - `concrete` does not exist
   - `terracotta` is listed twice

3. `calcite`
   - `grass_path` should be `dirt_path`
   - `hardened_clay` does not exist
   - `planks` does not exist
   - `silver_glazed_terracotta` does not exist
   - `stained_hardened_clay` does not exist

4. `coarse_dirt`
   - the line starting with `concrete=` does not seem to do anything
   - `grass_path` does not exist (`dirt_path` is already included)

5. `crimson_nylium`
   - `grass_path` should be `dirt_path`
   - `hardened_clay` does not exist
   - `planks` does not exist
   - `silver_glazed_terracotta` does not exist
   - `stained_hardened_clay` does not exist

6. `diorite`
   - `pointed_dripstone` does not seem to do anything

7. `rooted_dirt`
   - `grass_path` should be `dirt_path`
   - `hardened_clay` does not exist
   - `planks` does not exist
   - `stained_hardened_clay` does not exist

8. `grass_block`
   - `hardened_clay` does not exist
   - `planks` does not exist
   - `silver_glazed_terracotta` does not exist
   - `stained_hardened_clay` does not exist

9. `gravel`
   - `grass_path` should be `dirt_path`
   - `hardened_clay` does not exist
   - `planks` does not exist
   - `silver_glazed_terracotta` does not exist
   - `stained_hardened_clay` does not exist

10. `hay_block`
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `silver_glazed_terracotta` does not exist
    - `stained_hardened_clay` does not exist

11. `moss` / `moss_block`
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `stained_hardened_clay` does not exist

12. `mud`
    - does not have a `.properties` file

13. `mycelium`
    - `dirt` is listed twice
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `silver_glazed_terracotta` does not exist
    - `stained_hardened_clay` does not exist

14. `packed_mud`
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `stained_hardened_clay` does not exist

15. `podzol`
    - `dirt` is listed twice
    - `cobblestone_mossy_beach` does not exist
    - `cobblestone_mossy_birch` does not exist
    - `cobblestone_mossy_coldbeach` does not exist
    - `cobblestone_mossy_darkoak` does not exist
    - `cobblestone_mossy_desert` does not exist
    - `cobblestone_mossy_extremehills` does not exist
    - `cobblestone_mossy_forest` does not exist
    - `cobblestone_mossy_iceplains` does not exist
    - `cobblestone_mossy_jungle` does not exist
    - `cobblestone_mossy_mushroom` does not exist
    - `cobblestone_mossy_plains` does not exist
    - `cobblestone_mossy_swamp` does not exist
    - `cobblestone_mossy_taiga` does not exist
    - `concrete` does not exist
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `podzol` does not seem to do anything (same block)
    - `silver_glazed_terracotta` does not exist
    - `stained_hardened_clay` does not exist
    - `stonebrick` does not exist (`stone_bricks` is already included)
    - `cyan_terracotta` is listed twice
    - `dirt` is listed twice
    - `grass_path` is listed twice
    - `grass_path` should be `dirt_path`
    - `sandstone_stairs` do exist, but cause clipping; additionally, no other stairs are mentioned in `red_sand`
    - `silver_glazed_terracotta` does not exist

16. `reinforced_deepslate`
    - does not have a `.properties` file

17. `sand`
    - `dirt` is listed twice
    - `grass_path` is listed twice
    - `grass_path` should be `dirt_path`
    - `silver_glazed_terracotta` does not exist

18. `warped_nylium`
    - `dirt` is listed twice
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `silver_glazed_terracotta` does not exist
    - `stained_hardened_clay` does not exist

19. `amethyst_block` / `budding_amethyst`
    - `dirt` is listed twice
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `silver_glazed_terracotta` does not exist
    - `stained_hardened_clay` does not exist

20. `deepslate` / `deepslate_*_ore`
    - `1.properties` has empty `matchTiles=`
    - `20.properties` has empty `matchTiles=`

21. `dirt` / `rooted_dirt`
    - `grass_path` should be `dirt_path`
    - `hardened_clay` does not exist
    - `planks` does not exist
    - `stained_hardened_clay` does not exist

**Comment by xalixilax** [comment]
I had a plan to programmatically create my overlays but I had some work to do to parse every files I have currently. This should be really powerful et mitigate any current and upcoming issues. For your information, every texture/blocks mentioned are blocks that were present at a point. These are likely gonna stay to keep being compatible with older versions.

**Comment by Toatrika** [comment]
That makes sense to me. I'll try and fix the other stuff myself and submit a pull request (i have no idea how that works lol, but I'll figure it out)

**Comment by Toatrika** [comment]
actually, let me make this two separate issues because it kind of should be

**Comment by Rebel459** [comment]
Noticed a bunch of connected textures (mainly grass) weren't working, but only with Sodium for some reason @xalixilax

**Comment by Rebel459** [comment]
Though I suppose some overlays not working with sodium could be a separate issue?

**Comment by Toatrika** [comment]
> Though I suppose some overlays not working with sodium could be a separate issue?

yep, thats a separate issue. if you open this as a new issue you should ping/mention me or link it here, i could say some stuff thatd be off topic for this issue
